src/term/mod.rs

```diff
@@ -38,39 +39,78 @@ use self::cell::LineLength;
 pub struct RenderableCellsIter<'a> {
     grid: &'a mut Grid<Cell>,
     cursor: &'a Point,
+    cursor_type: CursorType,
     mode: TermMode,
     line: Line,
     column: Column,
     selection: Option<RangeInclusive<index::Linear>>,
+    cursor_cell: CursorCell,
+}
+
+enum CursorCell {
+    Extra(IndexedCell),
+    Replacement(IndexedCell),
 }
 
 impl<'a> RenderableCellsIter<'a> {
     /// Create the renderable cells iterator
     ///
     /// The cursor and terminal mode are required for properly displaying the
     /// cursor.
-    fn new<'b>(
-        grid: &'b mut Grid<Cell>,
-        cursor: &'b Point,
-        mode: TermMode,
-        selection: &Selection,
-    ) -> RenderableCellsIter<'b> {
+    fn new<'b>(grid: &'b mut Grid<Cell>,
+               cursor: &'b Point,
+               cursor_type: CursorType,
+               mode: TermMode,
+               selection: &Selection)
+               -> RenderableCellsIter<'b> {
         let selection = selection.span()
             .map(|span| span.to_range(grid.num_cols()));
 
+        let cursor_cell = CursorCell::Replacement(IndexedCell {
```

Comment:
Seems like this would be conditional on whether it's a block or something that goes on top of the character.